guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,750 @@
+to `compile()`.  The most popular Object Detection metrics are COCO metrics,
+which were published alongside the MSCOCO dataset.  KerasCV provides an easy
+to use suite of COCO metrics. under the `keras_cv.metrics.BoxCOCOMetrics`
+symbol:
+"""
+coco_metrics = keras_cv.metrics.BoxCOCOMetrics(
+    bounding_box_format="xywh", evaluate_freq=128
+)
+"""
+Due to the high computational cost of computing COCO metrics, the KerasCV
+`BoxCOCOMetrics` component requires an `evaluate_freq` parameter to be passed to
+its constructor.  Every `evaluate_freq`-th call to `update_state()`, the metric
+will recompute the result.  In between invocations, a cached version of the
+result will be returned.
+
+To force an evaluation, you may call `coco_metrics.result(force=True)`:
+"""
+model.compile(
+    classification_loss="focal",
+    box_loss="smoothl1",
+    optimizer=optimizer,
+    metrics=[coco_metrics],
+)
+result = model.evaluate(eval_ds)
+result = coco_metrics.result(force=True)
+print(tablify(result))
+
+"""
+**A note on TPU compatibility**
+
+Evaluation of `BoxCOCOMetrics` require running `tf.image.non_max_suppression()`
+inside of the `model.train_step()` and `model.evaluation_step()` functions.
+Due to this, the metric suite is not compatible with TPU when used with the
+`compile()` API.
+
+Luckily, there are two workarounds that allow you to still train a RetinaNet on TPU:
+
+- The use of a custom callback
+- Using a [SideCarEvaluator](https://www.tensorflow.org/api_docs/python/tf/keras/utils/SidecarEvaluator)
+
+Let's use a custom callback to achieve TPU compatibility in this guide:
+"""
+
+
+class EvaluateCOCOMetricsCallback(keras.callbacks.Callback):
+    def __init__(self, data):
+        super().__init__()
+        self.data = data
+        self.metrics = keras_cv.metrics.BoxCOCOMetrics(
+            bounding_box_format="xywh",
+            # passing 1e9 ensures we never evaluate until
+            # `metrics.result(force=True)` is
+            # called.
+            evaluate_freq=1e9,
+        )
+
+    def on_epoch_end(self, epoch, logs):
+        self.metrics.reset_state()
+        for batch in tqdm.tqdm(self.data):
+            images, y_true = batch[0], batch[1]
+            y_pred = self.model.predict(images, verbose=0)
+            self.metrics.update_state(y_true, y_pred)
+
+        metrics = self.metrics.result()
+        logs.update(metrics)
+        return logs
+
+
+"""
+## Training our model
+
+All that is left to do is train our model.  KerasCV object detection models follow the
+standard Keras workflow, leveraging `compile()` and `fit()`.
+
+Let's compile our model:
+"""
+
+model.compile(
+    classification_loss="focal",
+    box_loss="smoothl1",
+    optimizer=optimizer,
+    # We will use our custom callback to evaluate COCO metrics
+    metrics=None,
+)
+
+"""
+And run `model.fit()`!
+"""
+
+# If you want to train the fully model, uncomment `.take(20)` from each
```

Comment:
I would suggest showing how to instantiate a model from a pretrained backbone and then running a small number of steps without worrying about visualization and metric values. This is what most users will want to do and we want a "plug and play" example for them to copy.

Reply:
Last time we shipped the guide it felt pretty incomplete without a dedicated inference section.  

Personally, I say we keep it!